src/main/java/blink/Ui.java

```diff
@@ -0,0 +1,153 @@
+package blink;
+
+import blink.task.Task;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Ui that control logic of what is displayed on screen.
+ */
+public class Ui {
+
+    private static final String SPACING = "--------------------------------------";
+    private Scanner sc;
+
+    /**
+     * Constructor for Ui.
+     */
+    public Ui() {
+        this.sc = new Scanner(System.in);
+    }
+
+    /**
+     * Displays welcome message to user when Blink program starts.
+     */
+    public void showWelcome() {
+        System.out.println(Ui.SPACING + "\n"
+                + "Hello! Blink.Blink here\n"
+                + "What can I do for you today?\n"
+                + Ui.SPACING);
+    }
+
+    /**
+     * Display goodbye message when Blink program ends.
+     */
+    public void showBye() {
+        System.out.println("Bye bye~ Glad to be of service :D");
+    }
+
+    /**
+     * Displays the SPACING constant between lines.
+     */
+    public void showLine() {
+        System.out.println(Ui.SPACING);
+    }
+
+    /**
+     * Shows the user input error found.
+     *
+     * @param errMessage Error message of exceptions caught
+     */
+    public void showError(String errMessage) {
+        System.out.println("Error found: " + errMessage);
+    }
+
+    /**
+     * Reads the next line of user input.
+     *
+     * @return Next time of user input
+     */
+    public String readCommand() {
+        return sc.nextLine();
+    }
+
+    /**
+     * Display all the Tasks in TaskList.
+     *
+     * @param tasks TaskList which tasks are to be displayed.
+     */
+    public void showList(TaskList tasks) {
+        System.out.println(tasks.listTask());
+    }
+
+    /**
+     * Display the task to be marked.
+     *
+     * @param tasks TaskList of all the current Tasks
+     * @param num Number position of Task to mark
+     */
+    public void mark(TaskList tasks, int num) {
```

Comment:
Could this method be renamed? (i.e. showMarkedTask)
Likewise for the ones below.

Reply:
Yup, TaskList has functions with similar names so it might be hard for you at especially at a later stage.